x/protocolpool/keeper/genesis.go

```diff
@@ -0,0 +1,88 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/x/protocolpool/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) InitGenesis(ctx context.Context, data *types.GenesisState) error {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	// Perform data validation
+	if err := types.ValidateGenesis(data); err != nil {
```

Comment:
Seems redundant, aren't validating that at ValidateGenesis?

Reply:
oh, you're right
will remove unnecessary validation